From 5042694cebfa47cf47c14f66211f2229ebd1082a Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Sun, 24 Jan 2021 15:19:47 +0200
Subject: [PATCH 61/89] iot-gate-imx8: add support for the IE-TPM module

The CAN and the TPM modules are 2 alternatives in the baseboard
IED-BASE, slots A and B for the SPI bus.

Add 2 disabled nodes for the TPM modules.
Disable the CAN bus nodes.
The appropriate nodes will be enabled by the U-Boot, according to the
detected modules.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 .../boot/dts/compulab/sb-iotgimx8-ied.dts     | 50 +++++++++++++++++++
 arch/arm64/configs/iot-gate-imx8_defconfig    |  1 +
 2 files changed, 51 insertions(+)

diff --git a/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts
index 430561babc15..5e92ac790159 100644
--- a/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts
+++ b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied.dts
@@ -47,6 +47,26 @@
 		regulator-always-on;
 		enable-active-low;
 	};
+
+	regulatot-tpm0-rst {
+		compatible = "regulator-fixed";
+		regulator-name = "tpm0-rst";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&gpio4 31 GPIO_ACTIVE_HIGH>;
+		regulator-always-on;
+		enable-active-high;
+	};
+
+	regulator-tpm1-rst {
+		compatible = "regulator-fixed";
+		regulator-name = "tpm1-rst";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&gpio4 31 GPIO_ACTIVE_HIGH>;
+		regulator-always-on;
+		enable-active-high;
+	};
 };
 
 &i2c4 {
@@ -99,6 +119,15 @@
 		interrupts = <0 IRQ_TYPE_LEVEL_LOW>;
 		spi-max-frequency = <20000000>;
 		clocks = <&clk40m>;
+		status = "disabled";
+	};
+	tpm0: tpm@0 {
+		compatible = "infineon,slb9670", "tcg,tpm_tis-spi";
+		reg = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_tpm0>;
+		spi-max-frequency = <5000000>;
+		status = "disabled";
 	};
 };
 
@@ -120,6 +149,15 @@
 		interrupts = <29 IRQ_TYPE_LEVEL_LOW>;
 		spi-max-frequency = <20000000>;
 		clocks = <&clk40m>;
+		status = "disabled";
+	};
+	tpm1: tpm@0 {
+		compatible = "infineon,slb9670", "tcg,tpm_tis-spi";
+		reg = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_tpm1>;
+		spi-max-frequency = <5000000>;
+		status = "disabled";
 	};
 };
 
@@ -215,5 +253,17 @@
 				MX8MM_IOMUXC_UART4_RXD_GPIO5_IO28	0x00
 			>;
 		};
+
+		pinctrl_tpm0: tpm0grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SAI3_TXFS_GPIO4_IO31       0x0
+			>;
+		};
+
+		pinctrl_tpm1: tpm1grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_UART4_RXD_GPIO5_IO28	0x0
+			>;
+		};
 	};
 };
diff --git a/arch/arm64/configs/iot-gate-imx8_defconfig b/arch/arm64/configs/iot-gate-imx8_defconfig
index 912ac31d0093..f80375e87389 100644
--- a/arch/arm64/configs/iot-gate-imx8_defconfig
+++ b/arch/arm64/configs/iot-gate-imx8_defconfig
@@ -323,6 +323,7 @@ CONFIG_IPMI_HANDLER=m
 CONFIG_IPMI_DEVICE_INTERFACE=m
 CONFIG_IPMI_SI=m
 CONFIG_TCG_TPM=y
+CONFIG_TCG_TIS_SPI=m
 CONFIG_TCG_TIS_I2C_INFINEON=y
 CONFIG_I2C_CHARDEV=y
 CONFIG_I2C_MUX=y
-- 
2.17.1

